x/eibc/keeper/msg_server.go

```diff
@@ -23,25 +26,25 @@ var _ types.MsgServer = msgServer{}
 func (m msgServer) FulfillOrder(goCtx context.Context, msg *types.MsgFulfillOrder) (*types.MsgFulfillOrderResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 	logger := ctx.Logger()
-	// Check that the msg is valid
+
 	err := msg.ValidateBasic()
 	if err != nil {
 		return nil, err
 	}
-	// Check that the order exists in status PENDING
-	demandOrder, err := m.GetDemandOrder(ctx, commontypes.Status_PENDING, msg.OrderId)
+
+	demandOrder, err := m.GetMutableOrder(ctx, msg.OrderId)
```

Comment:
I understand the need to wrap the logic of checking if the demand order is relevant,  but I think that `Mutable` is confusing term here as it sounds like there are demand orders, which are active, but immutable - which is not the case.

"mutable" is simply pending and not fulfilled. 

maybe "active" is a better name (as we return error of `DemandOrderInactive` also) or another name which doesn't imply specific attirbute of a demand order.